src/super_gradients/training/utils/videos.py

```diff
@@ -0,0 +1,105 @@
+from typing import List, Optional, Tuple
+import cv2
+import os
+
+import numpy as np
+
+SUPPORTED_FORMATS = (".avi", ".mp4", ".mov", ".wmv", ".mkv")
+
+
+__all__ = ["load_video", "save_video"]
+
+
+def load_video(file_path: str, max_frames: Optional[int] = None) -> List[np.ndarray]:
+    """Open a video file and extract each frame into numpy array.
+
+    :param file_path:   Path to the video file.
+    :param max_frames:  Optional, maximum number of frames to extract.
+    :return:            Frames representing the video, each in (H, W, C).
+    """
+    cap = _open_video(file_path)
+    frames = _extract_frames(cap, max_frames)
+    cap.release()
+    return frames
+
+
+def _open_video(file_path: str) -> cv2.VideoCapture:
+    """Open a video file.
+
+    :param file_path:   Path to the video file
+    :return:            Opened video capture object
+    """
+    ext = os.path.splitext(file_path)[-1].lower()
+    if ext not in SUPPORTED_FORMATS:
```

Comment:
Well, supported formats is defined inside OpenCV library and why not we rely on `if not cap.isOpened()` only? This check looks unnecessary to me.

Reply:
I wasnt aware of that, thanks for that info